purldb-toolkit/src/purldb_toolkit/purlcli.py

```diff
@@ -123,21 +112,65 @@ def get_metadata_details(purls, output, file, unique, command_name):
         command_name=command_name,
         normalized_purls=normalized_purls,
         unique=unique,
+        purl_warnings=metadata_warnings,
     )
 
     return metadata_details
 
 
-def normalize_purl(purl):
+def check_metadata_purl(purl):
     """
-    Remove substrings that start with the '@', '?' or '#' separators.
+    Return a variable identifying the message for printing to the console by
+    get_metadata_details() if (1) the input PURL is invalid, (2) its type is not
+    supported by `metadata` or (3) its existence was not validated (e.g.,
+    "does not exist in the upstream repo").
+
+    This message will also be reported by construct_headers() in the
+    `warnings` field of the `header` section of the JSON object returned by
+    the `metadata` command.
     """
-    input_purl = purl
-    purl = purl.strip()
-    purl = re.split("[@,?,#,]+", purl)[0]
-    normalized_purl = purl
+    check_validation = validate_purl(purl)
+    if check_validation is None:
+        return "validation_error"
+    results = check_validation
+
+    if results["valid"] == False:
+        return "not_valid"
+
+    # This is manually constructed from a visual inspection of fetchcode/package.py.
+    metadata_supported_ecosystems = [
+        "bitbucket",
+        "cargo",
+        "github",
+        "npm",
+        "pypi",
+        "rubygems",
+    ]
+    metadata_purl = PackageURL.from_string(purl)
+
+    if metadata_purl.type not in metadata_supported_ecosystems:
+        return "valid_but_not_supported"
+
+    if results["exists"] == False:
+        return "not_in_upstream_repo"
+
+
+def normalize_purls(purls, unique):
```

Comment:
Add a docstring explaining this function

Reply:
Docstring added for `normalize_purls()`.